### DISCUSSION TITLE
Configure SSL certificate

### DISCUSSION THREAD

**Discussion by DhanasekarB** [Kong/kong#7189] [Help] [answered]
### Summary
- Kong version 2.3.2-alpine
- A concise summary of the bug.

Hi Team,

I Am trying to configure the SSL certificate in the latest version of kong. Seems like kong latest version has the default self signed certificate. need to override the certificate by using API/Custom configuration of kong.conf.default.

**API level:**
Curl -v -X POST http://localhost:8001/certificates
-F "cert=@test.crt"
-F "key=@test.key"
-F "snis[1]=example.com"   
and it's working fine for DNS "test-example.com" and it's not working for the actual IP address.

**For Example:**
curl -v -X GET https://test-example.com/ it returns our own certificate and looks good at the same time instead of DNS using 
IP address (curl -v -X GET https://xxx.xxx.xxx.xx/ )it returns the self-signed certificate.

how I override the default certificate?  Could you please help to configure the SSL certificate in the perfect way? 

**Custom level.**
I tried to configure the kong. conf.default  to  etc/kong/kong.conf

- Uncommented the SSL_cert and SSL_key in the kong.conf it's not working for me.

If possible could you please help how to configure the  docker file for the latest kong-version

**My DockerFiles:**
FROM kong:2.3.2-alphine
LABEL maintainer="test"
COPY ./config/kong.conf  /etc/kong

After building the image and then try to compose up the container and it's not working, it throws an error
**standard_init_linux.go:211: exec user process caused "no such file or directory"**

Please help to resolve the issue.

**Comment by kikito** [comment, marked as answer]
> For Example:
> curl -v -X GET test-example.com it returns our own certificate and looks good at the same time instead of DNS using
> IP address (curl -v -X GET xxx.xxx.xxx.xx )it returns the self-signed certificate.
>
> how I override the default certificate? Could you please help to configure the SSL certificate in the perfect way?

Hello, I am afraid I don't fully understand what you are trying to achieve. I suggest you rephrase by answering these questions, with as many details as possible:

> * What did you do?
>
> * What was the expected result?
>
> * What happened instead?


Also, please notice that the "latest" version of the Kong gateway is *2.4*, which was released recently. I recommend trying it, it might do what you need.
